File: place/src/main/java/pl/edu/pg/benchmarking/place/PlaceRepository.java
package pl.edu.pg.benchmarking.place;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.io.Reader;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.Optional;
import java.util.logging.Logger;

@Repository
public class PlaceRepository {

    private final Logger LOG = Logger.getLogger(this.getClass().getName());

    private List<Place> places;

    public List<Place> getPlaces() {
        return places;
    }

    public Optional<Place> findById(Integer id) {
        return places.stream().filter(place -> place.getId().equals(id)).findFirst();
    }

    @PostConstruct
    public void postConstruct() {
        try {
            Reader reader = Files.newBufferedReader(Paths.get("coordinates.json"));
            places = new Gson().fromJson(reader, new TypeToken<List<Place>>() {}.getType());
            reader.close();

        } catch (Exception ex) {
            LOG.severe("Can not read json file with cities coordinates");
        }
    }

}

File: Dotnet/monolith/Repositories/PlaceRepository.cs
using Monolith.Models;
using Monolith.Properties;

using System.Text.Json;

namespace Monolith.Repositories
{
    public class PlaceRepository
    {
        public List<Place> Places { get; private set; }

        public PlaceRepository()
        {
            Places = JsonSerializer.Deserialize<List<Place>>(Resources.coordinates);
        }

        public Place FindById(int id)
            => Places.FirstOrDefault(f => f.Id == id);
    }
}
